time/src/gloo.rs

```diff
@@ -0,0 +1,43 @@
+use std::{convert::TryInto, future::Future, pin::pin, time::Duration};
+
+use futures::future::{select, Either};
+use gloo_timers::future::{IntervalStream, TimeoutFuture};
+use send_wrapper::SendWrapper;
+
+pub type Interval = SendWrapper<IntervalStream>;
```

Comment:
From the [`send_wrapper` docs](https://docs.rs/send_wrapper/0.6.0/send_wrapper/index.html):
> This Rust library implements a wrapper type called `SendWrapper` which allows you to move around non-`Send` types between threads, as long as you access the contained value only from within the original thread.

This means we're locking our wasm implementation to one thread. I guess that's okay.